keps/sig-network/3070-reserved-service-ip-range/README.md

```diff
@@ -0,0 +1,436 @@
+
+#### Beta
+
+- Gather feedback from developers and community.
+- No issues reported.
+
+#### GA
+
+- 2 examples of real-world usage
+
+### Upgrade / Downgrade Strategy
+
+The feature only changes the allocation strategy, the bitmaps and the underlay logic remain the same,
+guaranteeing full compatibility and no risk rolling back the feature.
+
+### Version Skew Strategy
+
+Same as with upgrade, there is no change in the bitmaps, keeping the feature 100% compatible.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ServiceIPStaticSubrange
+  - Components depending on the feature gate: kube-apiserver
+- [x] Other
+  - Describe the mechanism:
+  - The feature require to restart the apiserver to use the new allocation strategy for Services, however,
+  it will not require downtime in multi-control-plane environments.
+    
+###### Does enabling the feature change any default behavior?
+
+Dynamic allocated IPs will not be chosen from the first 32 values of the Service IP range, that is transparent to the user,
+since the IP is and was already random, it doesn't matter from within the range the IP is obtained.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing, this only affects new requests and how these new request obtain the dynamically assigned IPs.
+
+###### Are there any tests for feature enablement/disablement?
+
+No need to add new test, current tests should keep working as today.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Current Services will not be affected, it doesn't matter in which range their IPs are allocated.
+This only affects the creation of new Services without an IP specified, there is no risk of impact on running workloads.
+
+###### What specific metrics should inform a rollback?
+
+The allocation logic already has the following metrics:
+
+  - allocated_ips
+  - available_ips
+  - allocation_total
+  - allocation_errors_total
+
+The increase of the errors metrics or the trend of the allocated_ips and available_ips doesn't change and new services are created or deleted.
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+### Monitoring Requirements
+
+Current allocator metrics are enough to understand the health of the allocation logic.
+
+  - allocated_ips
+  - available_ips
+  - allocation_total
+  - allocation_errors_total
+
+###### How can an operator determine if the feature is in use by workloads?
```

Comment:
I am embarrassed to not know this, but do we have a simple "settings" URL that lists things like this?

Reply:
Depending on the distro you can find the setting in different places, some use CRs to store the config, other store it in configmap , .... I don't know if there is some endpoint in the apiserver that gives you that, @liggitt may know better.

Reply:
I feel like we should have something, akin to metrics, which publishes static (or nearly static) settings, with a simple api like `settingz.AddInt("service-ip-static-subrange-size", 16)`

Reply:
we can publish an info metrics with the allocator information, same that apiserver does with the build info 

https://github.com/kubernetes/kubernetes/blob/55e1d2f9a73a0c78fc0c266236c3e31261635b90/staging/src/k8s.io/component-base/metrics/prometheus/version/metrics.go#L26-L33

WDYT?

Reply:
there's no /configz endpoint or equivalent. The things that crop up when this is requested for one-off things are:

1. who should have visibility to this
2. API structure and guarantees (e.g. should we communicate single configuration values or multiple? if multiple, in what format?)
3. behavior in multi-server clusters

Reply:
for the allocator I'll add the metrics with the constant  as in https://github.com/kubernetes/enhancements/pull/3071/files#r778133274 , the setting endpoint thing deserves its own kep :)